### PR TITLE
docs: use attributes in controllers page

### PR DESCRIPTION
**Pull request by alanpoulain**

### CODE REVIEW

**Review by dunglas — APPROVED** [review]
+1 but maybe should we just follow Symfony's best practices (even if I don't like this one) and extends `AbstractController` as well as changing `__invoke` to another name.
It will probably make this documentation easier to understand for users used to "the Symfony way".

WDYT?

File: core/controllers.md [hunk 1]
```diff
@@ -154,24 +156,20 @@ use ApiPlatform\Core\Annotation\ApiResource;
...
- *     }
- * })
- */
+#[ApiResource(itemOperations: [
+    'get',
+    'post_publication' => [
+        'method' => 'POST',
+        'path' => '/books/{id}/publication',
+        'controller' => CreateBookPublication::class,
+        'normalization_context' => ['groups' => ['publication']],
```
Comment (dunglas) [inline core/controllers.md:165]:
```suggestion
        'normalization_context' => ['groups' => 'publication'],
```

(syntactic sugar)

Comment (alanpoulain) [reply core/controllers.md:165]: Personally I prefer to always use arrays but OK.

**Comment by alanpoulain** [comment]
> +1 but maybe should we just follow Symfony's best practices (even if I don't like this one) and extends `AbstractController` as well as changing `__invoke` to another name.
> It will probably make this documentation easier to understand for users used to "the Symfony way".
> 
> WDYT?

Why not for the `AbstractController`. I'm not so sure about changing `__invoke` though, we would need to reference the method name in the `controller` attribute and we would lose the easy way to only use the FQCN.

**Pull request merged**